python_scripts/functions.py

```diff
@@ -269,11 +263,48 @@ def local_file_to_s3(local_path, s3_path):
     with open(local_path, "rb") as f:
         s3_client.upload_fileobj(f, b, o)
 
+
+
 def generate_iam_config(config, outpath="iam_config.yaml"):
     """
     Should take the necessary paths in the config and write out an
     iam that has access to those s3 paths e.g.
     write-only to log-base-path
     read/write to land-base-path
     """
-    pass
\ No newline at end of file
+
+    expected_keys = [
+        "iam-role-name",
+        "log-base-path",
+        "land-base-path",
+        "pass-base-path",
+        "fail-base-path"
+    ]
+
+    missing_keys = [key for key in expected_keys if key not in config]
+
+    if len(missing_keys) == 0:
+        out_iam = {
+            "iam-role-name": config["iam-role-name"],
+            "athena": {
+                "write": True
+            },
+            "s3": {
+                "write_only": [
+                    os.path.join(config['log-base-path'],'*')
+                ],
+                "read_write": [
+                    os.path.join(config['land-base-path'],'*'),
+                    os.path.join(config['fail-base-path'],'*'),
+                    os.path.join(config['pass-base-path'],'*'),
+                ]
+            }
+        }
+
+        with open(outpath, "w") as file:
+            yaml.dump(out_iam, file)
```

Comment:
Just wondering if we should give the function the option to writing out the actual IAM Policy using iam_builder?

```python
if outpath.endswith(".json"):
        iam = build_iam_policy(out_iam)
       with open(outpath, "w") as file:
            json.dump(iam, file, indent=4, separators=(',', ': '))
else:
    with open(outpath, "w") as file:
            yaml.dump(out_iam, file)
```